web/components/contract/contract-card.tsx

```diff
@@ -71,80 +87,233 @@ function getColor(contract: Contract) {
+    }
+  }
+
+  return (
+    <Col
+      className={clsx(
+        'relative -my-4 -mr-5 min-w-[6rem] justify-center gap-2 pr-5 pl-3 align-middle',
+        // Use this for colored QuickBet panes
+        // `bg-opacity-10 bg-${color}`
+        'bg-gray-50'
+      )}
+    >
+      <div>
+        <div
+          className="peer absolute top-0 left-0 right-0 h-[50%]"
+          onClick={() => placeQuickBet('UP')}
+        ></div>
+        <div className="mt-2 text-center text-xs text-transparent peer-hover:text-gray-400">
+          {formatMoney(10)}
+        </div>
+
+        {hasUpShares > 0 ? (
+          <TriangleFillIcon
+            className={clsx(
+              'mx-auto h-5 w-5',
+              `text-${color} text-opacity-70 peer-hover:text-gray-400`
+            )}
+          />
+        ) : (
+          <TriangleFillIcon className="mx-auto h-5 w-5 text-gray-200 peer-hover:text-gray-400" />
+        )}
+      </div>
+
+      <QuickOutcomeView contract={contract} />
+
+      <div>
+        <div
+          className="peer absolute bottom-0 left-0 right-0 h-[50%]"
+          onClick={() => placeQuickBet('DOWN')}
+        ></div>
+        {hasDownShares > 0 ? (
+          <TriangleDownFillIcon
+            className={clsx(
+              'mx-auto h-5 w-5',
+              `text-${color} text-opacity-70 peer-hover:text-gray-400`
+            )}
+          />
+        ) : (
+          <TriangleDownFillIcon className="mx-auto h-5 w-5 text-gray-200 peer-hover:text-gray-400" />
+        )}
+        <div className="mb-2 text-center text-xs text-transparent peer-hover:text-gray-400">
+          {formatMoney(10)}
+        </div>
+      </div>
+    </Col>
+  )
+}
+
+function QuickOutcomeView(props: { contract: Contract }) {
+  const { contract } = props
+  const { outcomeType } = contract
+  return (
+    <>
+      {outcomeType === 'BINARY' && (
+        <BinaryResolutionOrChance
+          className="items-center"
+          contract={contract}
+        />
+      )}
+
+      {outcomeType === 'NUMERIC' && (
+        <NumericResolutionOrExpectation
+          className="items-center"
+          contract={contract as NumericContract}
+        />
+      )}
+
+      {outcomeType === 'FREE_RESPONSE' && (
+        <FreeResponseResolutionOrChance
+          className="self-end text-gray-600"
+          contract={contract as FullContract<DPM, FreeResponse>}
+          truncate="long"
+        />
+      )}
+    </>
+  )
+}
+
 export function ContractCard(props: {
   contract: Contract
   showHotVolume?: boolean
   showCloseTime?: boolean
   className?: string
 }) {
-  const { contract, showHotVolume, showCloseTime, className } = props
+  const { showHotVolume, showCloseTime, className } = props
+  const [liveUpdate, setLiveUpdate] = useState(false)
+  const [contract, setContract] = useState<Contract>(props.contract)
   const { question, outcomeType } = contract
 
+  // When liveUpdate is true, start keeping this contract in sync
```

Comment:
This mostly works, but ESLint is complaining

```
246:6  Warning: React Hook useEffect has a missing dependency: 'contract'. Either include it or remove the dependency array. You can also do a functional update 'setContract(c => ...)' if you only need 'contract' in the 'setContract' call.
```

Any tips, React wizards?

Reply:
Outside the effect:
```const contractId = contract.id```
then,
```
useEffect(() => {
if (liveUpdate) {
      return listenForContract(contractId, (c) => c && setContract(c))
}, [contractId, liveUpdate])
```
Don't forget to return the listener so that it unsubscribes.

Reply:
But also, why not just always live update? Call useContract instead of all this.

(Will do a full review of this PR in a bit.)

Reply:
Lol I guess I assumed it was extra expensive to be listening to changes on all contract cards. Maybe not!

Reply:
I don't think it's that expensive! If they scroll down a bit, might fetch ~100 docs. If the markets are very active, could be more.

It's actually nice for the cards to show fresh data, too.